### PR TITLE
Feature implement digest email

### PR DESCRIPTION
**Pull request by KarenVentura**
@anymoto  @FerPerales 

This PR include
-Implement frequency digest email
-Change subscription into settings user
-Create rspec test for digest email
-Create design for digest email views

### CODE REVIEW

File: app/models/digest_builder.rb [hunk 1]
```diff
@@ -2,11 +2,28 @@ class DigestBuilder
...
+      DigestMailer.daily_digest(history, user.id).deliver!
     end
     DigestHistory.create_new!('daily')
   end
 
+  def self.send_weekly_email
+    history = DigestHistory.where(frequency: 'weekly').last || DigestHistory.create_new!('weekly')
+    User.subscriber_weekly_digest.each do |user|
+      DigestMailer.weekly_digest(history, user.id).deliver!
+      # DigestMailer.delay_until(7.days.from_now).weekly_digest(history, user.id)
```
Comment (FerPerales) [inline app/models/digest_builder.rb]: Remove this commented line if it is not needed anymore

File: app/models/digest_builder.rb [hunk 2]
```diff
@@ -2,11 +2,28 @@ class DigestBuilder
...
+    end
+    DigestHistory.create_new!('weekly')
+  end
+
+  def self.send_monthly_email
+    history = DigestHistory.where(frequency: 'monthly').last || DigestHistory.create_new!('monthly')
+    User.subscriber_monthly_digest.each do |user|
+      DigestMailer.monthly_digest(history, user.id).deliver!
+      # DigestMailer.delay_until(30.days.from_now).monthly_digest(history, user.id)
```
Comment (FerPerales) [inline app/models/digest_builder.rb]: Same here: please remove

File: app/views/digest_mailer/daily_digest.html.erb [hunk 3]
```diff
@@ -1,32 +1,70 @@
...
+            quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+            consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
+            cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
+            proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+          </tr>
+          <tr>
+            <h2 style="font-family: HelveticaNeue-Light; font-size: 18px; color: #F72D54; line-height: 24px; margin-left: 20px; padding: 20px;">Learning</h2><br>
+            <h2 style="font-family: HelveticaNeue-Bold;font-size: 15px;color: #6E4995;line-height: 18px; padding: 20px;">Build a game in 20 minutes.</h2><br>
+            <p style="font-family: HelveticaNeue;font-size: 12px;color: rgba(87,83,80,0.60);line-height: 15px; padding: 20px; "> Lorem ipsum dolor siamconsecteturadipisicing elit, sed do eiusmod
+            tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
```
Comment (FerPerales) [inline app/views/digest_mailer/daily_digest.html.erb]: Don't forget to update this part

File: app/views/layouts/one_column.html.erb [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
   	<title><%= content_for?(:title) ? yield(:title) : "RubyGameDev.com - A hub for discussion around Ruby game development" %></title>
 
-    <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.1/css/font-awesome.css" rel="stylesheet">
+    <!-- <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.1/css/font-awesome.css" rel="stylesheet"> -->
```
Comment (FerPerales) [inline app/views/layouts/one_column.html.erb:15]: Why is this line commented?

File: spec/models/digest_builder_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+describe DigestBuilder do
+
```
Comment (KarenVentura) [inline spec/models/digest_builder_spec.rb]: This test isn't necessary..

File: app/views/users/edit_current_user.html.erb [hunk 6]
```diff
@@ -1,24 +1,55 @@
...
+            <%= f.label :digest_email, t('digest_email.digest') %> <br/><br/>
+
+            <%= t('digest_email.updates_on') %> <br/>
+            <%= t('digest_email.when_certain') %> <br/>
+            <%= t('digest_email.email_update') %> <br/>
+
+            <%= f.label t('digest_email.digest') %> &nbsp;&nbsp;&nbsp;&nbsp;
+            <%= f.check_box(:digest_subscriber) %> <br/>
+
+            <%= f.radio_button(:digest_frequency, "daily", :class=> "radio-inline")%>
```
Comment (FerPerales) [inline app/views/users/edit_current_user.html.erb:36]: Please use new syntax for hash
class: "radio-inline"

File: app/workers/monthly_digest_worker.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  include Sidekiq::Worker
+
+  def perform
+    DigestBuilder.send_monthly_email
+  end
+  
+  # send email every first day in the month
+  Sidekiq::Cron::Job.create(
+  name: "Send Monthly digest email",
+  # cron: "0 0 1 * *",
```
Comment (FerPerales) [inline app/workers/monthly_digest_worker.rb]: Commented line

File: spec/factories/digest_history.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :digest_history do
+  	frequency "daily"
```
Comment (FerPerales) [inline spec/factories/digest_history.rb]: Please check this

File: spec/features/post_creation_spec.rb [hunk 9]
```diff
@@ -16,26 +16,26 @@
     click_button 'Create Post'
   end
 
-  it "shows a success message" do
+  xit "shows a success message" do
```
Comment (FerPerales) [inline spec/features/post_creation_spec.rb]: I don't think we should skip these specs...

File: spec/models/digest_builder_spec.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  
+  describe "#send_daily_email" do
+    let(:daily_digest_builder) { DigestBuilder.send_daily_email }
+  
+    it "send an email" do
+      expect(daily_digest_builder).to change { ActionMailer::Base.deliveries.count }.by(1)
+    end
+  end
+
+  describe "#send_weekly_email" do
```
Comment (FerPerales) [inline spec/models/digest_builder_spec.rb]: Specs missing for this

File: app/views/comments/_comment.html.erb [hunk 11]
```diff
@@ -1,4 +1,11 @@
 <div class="comment">
-  <p><a href="#" class="author"><%= comment.user.username %></a> - <time><%= time_ago_in_words comment.created_at %> ago</time></p>
-  <%= simple_format comment.body %>
+  <p>
+    <a href="#" class="author">
```
Comment (anymoto) [inline app/views/comments/_comment.html.erb]: let's use link_to helper instead

File: Gemfile [hunk 12]
```diff
@@ -56,6 +56,8 @@ gem 'ruby-stackoverflow', git: 'https://github.com/PowerCodeGirls/ruby-stackover
 
 gem 'sidekiq'
 gem "sidekiq-cron", "~> 0.3.0"
+gem 'rails-i18n', '~> 4.0.0'
+gem 'sinatra', :require => nil
```
Comment (KarenVentura) [inline Gemfile:60]: because we used it for monitoring sidekiq,

File: app/views/pages/home.html.erb [hunk 13]
```diff
@@ -1,7 +1,7 @@
 <% if signed_in? and !current_user.dismissed_welcome_message %>
   <div id="welcome_message" class="alert alert-success alert-dismissable">
     <button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button>
-    Welcome to RubyGameDev.com, the community-driven, information hub which helps others to build games in Ruby! First time here? <%= link_to 'Learn More', about_page_path, class: 'alert-link' %>.
+    <%= t('post.welcome') %><%= link_to 'Learn More', about_page_path, class: 'alert-link' %>.
```
Comment (anymoto) [inline app/views/pages/home.html.erb]: Why are we adding the work on Posts to the digest email PR?

File: app/views/layouts/one_column.html.erb (hunk 4 above)
Comment (sandrely2611-zz) [inline app/views/layouts/one_column.html.erb:15]: this line was commented before

File: app/views/digest_mailer/daily_digest.html.erb (hunk 3 above)
Comment (KarenVentura) [inline app/views/digest_mailer/daily_digest.html.erb]: :+1:

File: Gemfile (hunk 12 above)
Comment (anymoto) [inline Gemfile:60]: All right

**Comment by sandrely2611-zz** [comment]
@FerPerales

File: app/views/layouts/one_column.html.erb (hunk 4 above)
Comment (FerPerales) [inline app/views/layouts/one_column.html.erb:15]: ok :ok_hand:

File: spec/rails_helper.rb [hunk 14]
```diff
@@ -54,4 +54,12 @@
 
   config.include Capybara::DSL
   config.include Support::Features
+
+
+  Capybara.register_driver :webkit do |app|
```
Comment (FerPerales) [inline spec/rails_helper.rb]: We don't need this anymore

**Comment by FerPerales** [comment]
LGTM.If you can refactor the templaes for the digest email, that would be great :ok_hand:

**Pull request merged**